src/jit/morph.cpp

```diff
@@ -11571,7 +11571,18 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
             break;
 
         case GT_DIV:
-
+            /* Replace "val / dcon" with "val * (1.0 / dcon)" if dcon is a power of two */
```

Comment:
It would be good to give a brief explanation of why this only handles powers of two. Namely, powers of two within range are always exactly represented, so multiplication by the reciprocal is safe in this scenario.